src/query/execute_statement.c

```diff
@@ -18148,3 +18212,562 @@ get_dblink_info_from_dbserver (PARSER_CONTEXT * parser, const char *server, DB_V
 
   return error;
 }
+
+
+#define DBLINK_PASSWORD_MAX_LENGTH      (128)
+#define DBLINK_PASSWORD_CONFUSED_LENGTH (6)	// include 4(int) + 1(unsigned char) +  1(unsigned char)
+// Valid data size is the largest multiple of 3 less than or equal to DBLINK_PASSWORD_CIPHER_LENGTH.
+#define DBLINK_PASSWORD_CIPHER_LENGTH   (DBLINK_PASSWORD_MAX_LENGTH + DBLINK_PASSWORD_CONFUSED_LENGTH)
+#define DBLINK_PASSWORD_PAD_LENGTH      (40)	// include 2(length) + 2(mk) + 2(length) + 32(mk), Must be 4 or more
+#define DBLINK_PASSWORD_MAX_BUFSIZE  ((int)(DBLINK_PASSWORD_CIPHER_LENGTH / 3 * 4) + DBLINK_PASSWORD_PAD_LENGTH)
+
+/*
+ * pt_check_dblink_password ()  : Check the validity of the entered password.
+ *
+ * return		  : NO_ERROR or error code.
+ * parser(in)		  : Parser context.
+ * passwd(in)             : Password entered by user(or parser)
+ * cipher_buf(out)	  : Encrypted password
+ * ciper_buf_size(in)	  : Size of cipher_buf
+ * 
+ * Remark: 
+ *     Checks whether the entered password is a raw password or an encrypted password,
+ *     if it is a raw password, encrypt it.
+ *     The length of the raw password is shorter than DBLINK_PASSWORD_MAX_LENGTH.
+ *     If it is an encrypted password, it must have a specific length.
+ *     Even if there is no raw password, it will be returned as an encrypted password.
+ */
+int
+pt_check_dblink_password (PARSER_CONTEXT * parser, const char *passwd, char *cipher_buf, int ciper_buf_size)
+{
+  DB_VALUE val;
+  char *str;
+  int max_len = DBLINK_PASSWORD_MAX_BUFSIZE;
+  int err = ER_FAILED;
+  int length = 0;
+
+  /* Adjust the length so that it is a multiple of 4. */
+  max_len >>= 2;
+  max_len <<= 2;
+
+  if (ciper_buf_size <= max_len)
+    {
+      return ER_FAILED;
+    }
+
+  if (passwd && *passwd)
+    {
+      length = strlen (passwd);
+    }
+
+  db_make_null (&val);
+  if (length <= DBLINK_PASSWORD_MAX_LENGTH)
```

Comment:
When are raw password and encrypted password entered respectively? Can't the function be separated?

Reply:
The very first user input to perform the CREATE SERVER syntax raw Password,
Subsequent HA logs or passwords created by the unloaddb tool are encrypted.